src/luaclang.c

```diff
@@ -3,12 +3,146 @@
 #include "lualib.h"
 #include "lauxlib.h"
 
+#define INDEX  "ClangIndex"
+#define TU     "ClangTU"
+#define CURSOR "ClangCursor"
+#define TYPE "ClangType"
+
+static CXIndex* new_CXIndex(lua_State *L) 
+{
+        CXIndex *idx = (CXIndex*) lua_newuserdata(L, sizeof(CXIndex));
+        luaL_getmetatable(L, INDEX);
+        lua_setmetatable(L, -2);
+        return idx;
+}
+
+static CXIndex to_CXIndex(lua_State *L, int n)
+{
+        CXIndex* idx = (CXIndex*) luaL_checkudata(L, n, INDEX);
+        return *idx;
+}
+
+static CXTranslationUnit* new_CXTU(lua_State *L)
+{
+        CXTranslationUnit* tu = (CXTranslationUnit*) lua_newuserdata(L, sizeof(CXTranslationUnit));
+        luaL_getmetatable(L, TU);
+        lua_setmetatable(L, -2);
+        return tu;
+}
+
+static CXTranslationUnit to_CXTU(lua_State *L, int n) 
+{
+        CXTranslationUnit* tu = (CXTranslationUnit*) luaL_checkudata(L, n, TU);
+        return *tu;
+}
+
+static CXCursor* new_CXCursor(lua_State *L) 
+{
+        CXCursor* cur = (CXCursor*) lua_newuserdata(L, sizeof(CXCursor));
+        luaL_getmetatable(L, CURSOR);
+        lua_setmetatable(L, -2);
+        return cur;
+}
+
+static CXCursor to_CXCursor(lua_State *L, int n) 
+{
+        CXCursor* c = (CXCursor*) luaL_checkudata(L, n, CURSOR);
+        return *c;
+}
+
+static CXType* new_CXType(lua_State *L)
+{
+        CXType* t = (CXType*) lua_newuserdata(L, sizeof(CXType));
+        luaL_getmetatable(L, TYPE);
+        lua_setmetatable(L, -2);
+        return t;
+}
+
+
+/* Clang function */
+
+static int create_CXIndex(lua_State *L) 
+{
+        int exclude_pch = lua_toboolean(L, 1);
+        int diagnostics = lua_toboolean(L, 2);
+        CXIndex* idx = new_CXIndex(L);
+        *idx = clang_createIndex(exclude_pch, diagnostics);
+        return 1;
+}
+
+
+/* Index functions */
+
+static int dispose_CXIndex(lua_State *L) 
+{
+        CXIndex idx = to_CXIndex(L, 1);
+        clang_disposeIndex(idx);
+        return 0;
+}
+
+static int parse_TU(lua_State *L)
```

Comment:
You should create documentation for each function exported to Lua.
In this documentation you should describe the parameters and link to the clang documentation.
You don't need to explain what Clang itself does.